### PR TITLE
i18n(es): add `core-concepts/framework-components`

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [x] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

- adds `core-concepts/framework-components.md` in Spanish
- update right-sidebar ui
- adds page to nav

### CODE REVIEW

File: src/pages/es/core-concepts/framework-components.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+});
+```
+
+⚙️ Consulte la [guía de integraciones](/es/guides/integrations-guide/) para obtener más informarción sobre la instalación y configuración de las integraciones de Astro.
+
+⚙️ ¿Quieres ver un ejemplo del framework de tu elección? Visite [astro.new](https://astro.new/) y seleccione la plantilla de framework correspondiente.
+
+## Usando componentes del framework
+
+¡Use sus componentes de framework en sus páginas, plantillas y componentes de Astro como su fueran componentes de Astro! Todos sus componentes pueden vivir juntos en `/src/components`, o pueden organizarse de la forma que desee.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:45]:
```suggestion
¡Use sus componentes de framework en sus páginas, plantillas y componentes de Astro como si fueran componentes de Astro! Todos sus componentes pueden vivir juntos en `/src/components`, o pueden organizarse de la forma que desee.
```

File: src/pages/es/core-concepts/framework-components.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+> 💡 _Recuerde: ¡todas las importaciones deben vivir en la **parte superior** del script de su componente de Astro!_
+
+De forma predeterminada, sus componentes de framework se renderizarán como HTML estático. Esto es útil para crear maquetados de componentes que no son interactivos y evita enviar código JavaScript innecesario al cliente.
+
+## Hidratando componentes interactivos
+
+Un componente de framework puede hacerse interactivo (hidratado) usando una de las directivas `client:*`. Este es un atributo del componente para definir cómo se debe **renderizar** e **hidratar** su componente.
+
+Esta [directiva del cliente](/es/reference/directives-reference/#client-directives) describe si su componente se debe renderizar o no al momento de la compilación, además de cuando el navegador debe cargar el JavaScript del lado del cliente de su componente.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:69]:
```suggestion
Esta [directiva del cliente](/es/reference/directives-reference/#client-directives) describe si su componente se debe renderizar o no al momento de la compilación, además de cuándo el navegador debe cargar el JavaScript del lado del cliente de su componente.
```

File: src/pages/es/core-concepts/framework-components.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+Most directives will render the component on the server at build time. Component JS will be sent to the client according to the specific directive. The component will hydrate when its JS has finished importing.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:71]: Seems that you forgot to delete this 😝

File: src/pages/es/core-concepts/framework-components.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+La mayoría de las directivas renderizarán el componente en el servidor al momento de la compilación. El componente de JavaScript se enviará al cliente de acuerdo a la directiva específicada. El componente se hidratará cuando su JavaScript haya terminado de importarse.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:73]:
```suggestion
La mayoría de las directivas renderizarán el componente en el servidor al momento de la compilación. El JavaScript del componente se enviará al cliente de acuerdo a la directiva especificada. El componente se hidratará cuando su JavaScript haya terminado de importarse.
```

File: src/pages/es/core-concepts/framework-components.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+---
+// Ejemplo: hidratando los componentes de framework en el navegador.
+import InteractiveButton from '../components/InteractiveButton.jsx';
+import InteractiveCounter from '../components/InteractiveCounter.jsx';
+---
+<!-- Este componente de JavaScript comenzará a importarse cuando se cargue la página-->
+<InteractiveButton client:load />
+
+<!-- El JavaScript de este componente no se enviará al cliente hasta que
+el usuario se desplaze hacia abajo y el componente sea visible en la página -->
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:85]:
```suggestion
el usuario se desplace hacia abajo y el componente sea visible en la página -->
```

File: src/pages/es/core-concepts/framework-components.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+<InteractiveCounter client:visible />
+```
+
+> ⚠️ Cualquier renderizador de JavaScript necesario para el componente de framework (por ejemplo, React, Svelte) se descargará con la página. Las directivas `client:*` solo dictan cuándo se importa el _componente de JavaScript_ y cuándo se hidrata el _componente_.
+
+### Directivas de hidratación disponibles
+
+Hay varias directivas de hidratación disponibles para los componentes del framework: `client:load`, `client:idle`, `client:visible`, `client:media={QUERY}` y `client:only={FRAMEWORK}`.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:93]:
```suggestion
Hay varias directivas de hidratación disponibles para los componentes de framework: `client:load`, `client:idle`, `client:visible`, `client:media={QUERY}` y `client:only={FRAMEWORK}`.
```

File: src/pages/es/core-concepts/framework-components.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+</MyReactSidebar>
+```
+
+> ⚠️ *Recuerde: los propios archivos de los componentes de framework (por ejemplo, `.jsx`, `.svelte`) no pueden combinar varios frameworks.*
+
+Esto le permite crear "aplicaciones" completas en su framework de JavaScript preferido y renderizarlas, a través de un componente principal, en una página de Astro. Este es un patrón conveniente para permitir que los componentes relacionados compartan estados o contextos.
+
+Cada framework tiene sus propios patrones para anidar: `children` props y [render props](https://reactjs.org/docs/render-props.html) para React y Solid; `<slot />` con o sin nombres para Svelte y Vue, por ejemplo.
+
+Nota: los componentes de Astro siempre se renderizan en HTML estático, incluso cuando incluyen componentes de framework que son hidratados. Esto significa que solo se pueden pasar props que no hacen ninguna renderización a HTML. Pasar los "render props" de React o los slots con nombre a los componentes del framework desde un componente de Astro no funcionará porque los componentes de Astro no pueden proporcionar la ejecución del cliente que requieren esos patrones.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:141]:
```suggestion
Nota: los componentes de Astro siempre se renderizan en HTML estático, incluso cuando incluyen componentes de framework que son hidratados. Esto significa que solo se pueden pasar props que no hacen ninguna renderización a HTML. Pasar los "render props" de React o los slots con nombre a los componentes de framework desde un componente de Astro no funcionará porque los componentes de Astro no pueden proporcionar la ejecución del cliente que esos patrones requieren.
```

File: src/pages/es/core-concepts/framework-components.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+## ¿Puedo hidratar los componentes de Astro?
+
+Si intentas hidratar un componente Astro con un modificador `cliente:`, obtendrás un error.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:145]:
```suggestion
Si intentas hidratar un componente Astro con un modificador `client:`, obtendrás un error.
```

File: src/pages/es/core-concepts/framework-components.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+Los [componentes de Astro](/es/core-concepts/astro-components/) son componentes de solo maquetado a HTML sin ninguna ejecución del lado del cliente. Pero puede usar una etiqueta `<script>` en el maquetado del componente Astro para enviar JavaScript al navegador que se ejecuta en el ámbito global.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:147]:
```suggestion
Los [componentes de Astro](/es/core-concepts/astro-components/) son componentes de maquetado únicamente a HTML sin ninguna ejecución del lado del cliente. Pero puede usar una etiqueta `<script>` en el maquetado del componente Astro para enviar JavaScript al navegador que se ejecuta en el ámbito global.
```

File: src/pages/es/core-concepts/framework-components.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+📚 Obtenga más información sobre [`<scripts>` del lado del cliente en los componentes de Astro](/es/core-concepts/astro-components/#scripts-del-lado-del-cliente)
+
+[mdn-io]: https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
+[mdn-ric]: https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback
+[mdn-mm]: https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:153]: We could change these 3 URLs for the spanish version, changing `en-US` for `es`, what do you think?

File: src/pages/es/core-concepts/framework-components.md [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+export default defineConfig({
+	integrations: [react(), preact(), svelte(), vue(), solid() , lit()],
+});
+```
+
+⚙️ Consulte la [guía de integraciones](/es/guides/integrations-guide/) para obtener más informarción sobre la instalación y configuración de las integraciones de Astro.
+
+⚙️ ¿Quieres ver un ejemplo del framework de tu elección? Visite [astro.new](https://astro.new/) y seleccione la plantilla de framework correspondiente.
+
+## Usando componentes del framework
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/framework-components.md:43]:
```suggestion
## Usando componentes de framework
```

**Review by agustinmulet — COMMENTED** [review]
Mostly typos to fix, after this is fixed => LGTM!

**Comment by kevinzunigacuellar** [comment]
ready to merge @delucis 🚀

**Pull request merged**